nixos/modules/services/networking/opengfw.nix

```
@@ -0,0 +1,353 @@
+            };
+
+            modifier = mkOption {
+              default = null;
+              description = ''
+                Modification of specified packets when using the `modify` action. [Available modifiers](https://github.com/apernet/OpenGFW/tree/master/modifier)
+              '';
+              type = types.nullOr (
+                types.submodule {
+                  options = {
+                    name = mkOption {
+                      description = "Name of the modifier.";
+                      type = types.singleLineStr;
+                      example = "dns";
+                    };
+
+                    args = mkOption {
+                      description = "Arguments passed to the modifier.";
+                      type = types.attrs;
+                      example = {
+                        a = "0.0.0.0";
+                        aaaa = "::";
+                      };
+                    };
+                  };
+                }
+              );
+            };
+          };
+        }
+      );
+
+      example = [
+        {
+          name = "block v2ex http";
+          action = "block";
+          expr = ''string(http?.req?.headers?.host) endsWith "v2ex.com"'';
+        }
+        {
+          name = "block google socks";
+          action = "block";
+          expr = ''string(socks?.req?.addr) endsWith "google.com" && socks?.req?.port == 80'';
+        }
+        {
+          name = "v2ex dns poisoning";
+          action = "modify";
+          modifier = {
+            name = "dns";
+            args = {
+              a = "0.0.0.0";
+              aaaa = "::";
+            };
+          };
+          expr = ''dns != nil && dns.qr && any(dns.questions, {.name endsWith "v2ex.com"})'';
+        }
+      ];
+    };
+  };
+
+  config = mkIf cfg.enable {
+    security.wrappers.OpenGFW = {
+      owner = cfg.user;
+      group = cfg.user;
+      capabilities = "cap_net_admin+ep";
+      source = "${cfg.package}/bin/OpenGFW";
+    };
+
+    systemd.services.opengfw = {
+      description = "OpenGFW";
+      wantedBy = ["multi-user.target"];
+      after = ["network.target"];
+      path = with pkgs; [iptables];
+      preStart = ''
+        ${optionalString (rules != null) "ln -sf ${rules} rules.yaml"}
+        ${optionalString (settings != null) "ln -sf ${settings} config.yaml"}
+      '';
+
+      serviceConfig = rec {
+        WorkingDirectory = cfg.dir;
+        ExecStart = "${config.security.wrapperDir}/OpenGFW -f ${cfg.logFormat} -l ${cfg.logLevel} -c config.yaml rules.yaml ";
+        ExecReload = "kill -HUP $MAINPID";
+        Restart = "always";
+        User = cfg.user;
+        StandardOutput = mkIf (cfg.logFile != null) "append:${cfg.logFile}";
+        StandardError = StandardOutput;
+      };
+    };
+
+    users = {
+      groups.${cfg.user} = {};
+      users.${cfg.user} = {
+        description = "opengfw user";
+        isNormalUser = true;
```

Not sure why this is a normal user.
```suggestion
        isSystemUser = true;
```